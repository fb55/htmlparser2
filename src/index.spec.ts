import {
    parseDocument,
    parseDOM,
    createDomStream,
    DomHandler,
    DefaultHandler,
} from "./index.js";
import { Element } from "domhandler";

// Add an `attributes` prop to the Element for now, to make it possible for Jest to render DOM nodes.
Object.defineProperty(Element.prototype, "attributes", {
    get() {
        return Object.keys(this.attribs).map((name) => ({
            name,
            value: this.attribs[name],
        }));
    },
    configurable: true,
    enumerable: false,
});

describe("Index", () => {
    test("parseDocument", () => {
        const dom = parseDocument("<a foo><b><c><?foo>Yay!");
        expect(dom).toMatchSnapshot();
    });

    test("parseDOM", () => {
        const dom = parseDOM("<a foo><b><c><?foo>Yay!");
        expect(dom).toMatchSnapshot();
    });

    test("createDomStream", (done) => {
        const domStream = createDomStream((error, dom) => {
            expect(error).toBeNull();
            expect(dom).toMatchSnapshot();

            done();
        });

        for (const c of "&amp;This is text<!-- and comments --><tags>") {
            domStream.write(c);
        }

        domStream.end();
    });

    describe("API", () => {
        it("should export the appropriate APIs", () => {
            expect(DomHandler).toEqual(DefaultHandler);
        });
    });
});
