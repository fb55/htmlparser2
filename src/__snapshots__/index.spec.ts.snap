// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index createDomStream 1`] = `
[
  &This is text,
  <!-- and comments -->,
  <tags />,
]
`;

exports[`Index parseDOM 1`] = `
[
  <a
    foo=""
  >
    <b>
      <c>
        ProcessingInstruction {
          "data": "?foo",
          "endIndex": null,
          "name": "?foo",
          "next": Yay!,
          "parent": <c>
            [Circular]
            Yay!
          </c>,
          "prev": null,
          "startIndex": null,
          "type": "directive",
        }
        Yay!
      </c>
    </b>
  </a>,
]
`;

exports[`Index parseDocument 1`] = `
Document {
  "children": [
    <a
      foo=""
    >
      <b>
        <c>
          ProcessingInstruction {
            "data": "?foo",
            "endIndex": null,
            "name": "?foo",
            "next": Yay!,
            "parent": <c>
              [Circular]
              Yay!
            </c>,
            "prev": null,
            "startIndex": null,
            "type": "directive",
          }
          Yay!
        </c>
      </b>
    </a>,
  ],
  "endIndex": null,
  "next": null,
  "parent": null,
  "prev": null,
  "startIndex": null,
  "type": "root",
}
`;
